src/librustc_resolve/lib.rs

```diff
@@ -444,9 +443,15 @@ fn resolve_error<'b, 'a: 'b, 'tcx: 'a>(resolver: &'b Resolver<'a, 'tcx>,
                                                    module = &*path,
                                                    ident = ident.node);
                             }
-
-                            _ => {} // no help available
+                            ExprCall(_, _) => {
+                                help_msg = format!("No function corresponds to `{module}(..)`",
+                                                   module = &*path);
+                            }
+                            _ => { } // no help available
                         }
+                    } else {
+                        help_msg = format!("Module `{module}` cannot be used as an object!",
```

Comment:
Okay, so I *don't* like the phrasing "used as an object" here.

I would instead say something like "Module `{module}` cannot be passed around as a first-class value."

(except that may be a bit long winded... maybe just "Module `{module}` cannot be the value of an expression" ?)